distributed/scheduler.py

```diff
@@ -279,7 +279,10 @@ def __hash__(self):
         return hash((self.name, self.host))
 
     def __eq__(self, other):
-        return type(self) == type(other) and hash(self) == hash(other)
+        return type(self) == type(other) and (self.name, self.host) == (
+            other.name,
+            other.host,
+        )
```

Comment:
I am wondering if using `name` and `host` is correct here:
* `name` is optional and could be None
* `host` is not necessary unique as multiple workers could be running on the same machine

I am lacking necessary Dask context, so I am wondering why you have not picked `address` instead? It is used for indexing the `WorkerState` and the documentation says it is `This worker's unique key.`.

Reply:
I was just pulling items from the `identity()` function, no personal reason to prefer one to the other. @TomAugspurger @mrocklin would you agree `address` makes more sense here?